arbor/backends/gpu/stack.hpp

```diff
@@ -67,41 +81,67 @@ class stack {
         storage_(create_storage(capacity)), gpu_context_(gpu_ctx) {}
 
     ~stack() {
-        storage_.synchronize();
-        if (storage_->data) {
-            allocator<value_type>().deallocate(storage_->data, storage_->capacity);
+        memory::cuda_sync();
```

Comment:
this synchronization is not needed (why perform it here, and not at other times before performing copy from device to host?)

Reply:
Removed.

Reply:
Removed.